### PR TITLE
Refactor 3

### PR DESCRIPTION
**Pull request by d-atkins**
## Description
Bug fix/refactors

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which maintains existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
- Adds validation for review ratings
- Fix bug where deleting a shelter wouldn't delete any of those shelter's pets from favorites
- Fix bug with flash message persisting on failed shelter creation
- Make review failure feedback specific
- Updates style

## RSpec Results
```
Finished in 5.01 seconds (files took 2.1 seconds to load)
111 examples, 0 failures

Coverage report generated for RSpec - 1376 / 1376 LOC (100.0%) covered.
```

### CODE REVIEW

File: app/controllers/favorites_controller.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def update
   end
 
   def index
-    @pets = Pet.find(favorites.contents)
+    @pets = Pet.where(id: favorites.contents)
```
Comment (d-atkins) [inline app/controllers/favorites_controller.rb]: should make it so if someone else deletes a pet that's in your favorites, the page can still render

File: app/controllers/shelters_controller.rb [hunk 2]
```diff
@@ -39,6 +39,7 @@ def update
   def destroy
     shelter = Shelter.find(params[:id])
     if shelter.pets.pending.empty?
+      favorites.delete_pets(shelter.pets.pluck(:id))
```
Comment (d-atkins) [inline app/controllers/shelters_controller.rb]: delete all favorited pets from this shelter before deleting it

File: app/models/review.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class Review < ApplicationRecord
   validates_presence_of :title, :content, :rating
-  belongs_to :shelter, dependent: :destroy 
+  validates :rating, numericality: {less_than_or_equal_to: 5, greater_than_or_equal_to: 0}
```
Comment (d-atkins) [inline app/models/review.rb]: only allows ints greater than or equal to 0 and less than or equal to 5

File: app/controllers/shelters_controller.rb (hunk 2 above)
Comment (DavidHoltkamp1) [reply app/controllers/shelters_controller.rb]: Thought this would be a good way to do it! nice.

**Pull request merged**